src/parser.ml

```diff
@@ -1,5 +1,95 @@
 open Ast
 
+module Compat : sig
+  val rep: Uchar.t
+  (** Uchar.rep *)
+
+  val add_utf_8_uchar : Buffer.t -> Uchar.t -> unit
+  (** Buffer.add_utf_8_uchar *)
+
+  val find_opt: ('a -> bool) -> 'a list -> 'a option
+  (** List.find_opt *)
```

Comment:
The way these kinds of shims are usually done is to define a module `Compat` with submodules matching those of the stdlib:
```
module Compat = struct
  module Uchar = struct
    include Uchar
    let rep = ...
  end
  module Buffer = struct
    include Buffer
    let add_utf_8_uchar ... = ...
  end
  ...
end
```
and then simply `open Compat` before the affected code. That way there is no need to touch the actual code that makes use of these functions.

Reply:
Ah makes sense! Will do!